lib/paper_trail/version_concern.rb

```diff
@@ -329,13 +333,29 @@ def object_changes_deserialized
     # Enforces the `version_limit`, if set. Default: no limit.
     # @api private
     def enforce_version_limit!
-      limit = PaperTrail.config.version_limit
+      limit = version_limit
       return unless limit.is_a? Numeric
       previous_versions = sibling_versions.not_creates.
         order(self.class.timestamp_sort_order("asc"))
       return unless previous_versions.size > limit
       excess_versions = previous_versions - previous_versions.last(limit)
       excess_versions.map(&:destroy)
     end
+
+    # See docs section 2.e. Limiting the Number of Versions Created.
+    # The version limit can be global or per-model.
+    #
+    # @api private
+    #
+    # TODO: Duplication: similar `constantize` in Reifier#version_reification_class
+    def version_limit
+      if PaperTrail::Version.column_names.include?("item_subtype")
```

Comment:
```suggestion
      if self.class.item_subtype_column_present?
```
I think this is a must, now that you've added  the class method to check for the item_subtype...